src/DistributedLock.Postgres/PostgresDistributedLock.Extensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Medallion.Threading.Internal;
+using System.Data;
+
+namespace Medallion.Threading.Postgres;
+
+public partial class PostgresDistributedLock
+{
+    public static bool TryAcquireWithTransaction(PostgresAdvisoryLockKey key, IDbTransaction transaction, TimeSpan timeout = default, CancellationToken cancellationToken = default)
+    {
+        if (key == null) { throw new ArgumentNullException(nameof(key)); }
+        if (transaction == null) { throw new ArgumentNullException(nameof(transaction)); }
+
+        var connection = new PostgresDatabaseConnection(transaction);
+
+        var handle = DistributedLockHelpers.TryAcquire(PostgresAdvisoryLock.ExclusiveLock, connection, key.ToString(), timeout, cancellationToken);
+
+        return handle != null;
```

Comment:
We can use the `SyncViaAsync` class to unify the sync/async implementations here.

Reply:
Sure, done